app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt

```diff
@@ -231,7 +235,24 @@ class ApplicationModule(private val application: Application) {
 
     @Provides
     @Singleton
-    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage = AndroidKeyStorage(context)
+    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage =
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) {
+            val localeCopy = Locale.getDefault()
+            setLocale(context, Locale.ENGLISH)
+            val androidKeyStorage = AndroidKeyStorage(context)
+            setLocale(context, localeCopy)
+            androidKeyStorage
+        } else {
+            AndroidKeyStorage(context)
+        }
+
+    private fun setLocale(context: Context, locale: Locale) {
```

Comment:
maybe we could have a `private fun Context.setLocale()` method instead? Although it is not important.